src/RahaFirestore.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * A module that exports an api for interacting with Raha's Firestore db.
+ *
+ * Example usage:
+ * async function getAppliedOperations() {
+ *     return await get(
+ *         operationsCollectionFilters.applied(true)(
+ *             operationsCollection()));
+ * }
+ */
+
+import * as firebase from 'firebase';
+import 'firebase/firestore';
+
+let db;
+
+/**
+ * Retrieve an instance of the Firestore db.
+ */
+function getDb() {
+    if (!db) {
+        db = firebase.initializeApp(require('./config/firebase.config.json')).firestore();
+    }
+    return db;
+}
+
+/**
+ * Resolve a query on a Firestore collection.
+ */
+async function get(collection: firebase.firestore.CollectionReference) {
```

Comment:
Seems like a very "thin" function (TM - https://web.stanford.edu/~ouster/cgi-bin/cs190-winter18/reviewCodes.php)

Reply:
ahh, but it gives the whole module such a nice functional feel :D